### PR TITLE
Bundle processing refactoring & initial Ivy i18n infrastructure

### PR DESCRIPTION
**Pull request by clydin**

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/browser/index.ts [hunk 1]
```diff
@@ -162,18 +162,61 @@ function getCompilerConfig(wco: WebpackConfigOptions): webpack.Configuration {
...
+  if (
+    metadata.locales !== undefined &&
+    (!metadata.locales || typeof metadata.locales !== 'object' || Array.isArray(metadata.locales))
+  ) {
+    throw new Error('Project i18n locales field is malformed.  Expected an object.');
+  } else if (metadata.locales) {
+    for (const [locale, translationFile] of Object.entries(metadata.locales)) {
+      if (typeof translationFile !== 'string') {
+        throw new Error(
+          `Project i18n locales field value for '${locale}' is malformed.  Expected an string.`,
```
Comment (alan-agius4) [inline packages/angular_devkit/build_angular/src/browser/index.ts]: `Expected an string` -> `Expected a string`.

Also before `Expected` there seems to be a double space. Same for `Expected an object `

File: package.json [hunk 2]
```diff
@@ -87,6 +87,7 @@
     "@bazel/jasmine": "0.37.1",
     "@bazel/karma": "0.37.1",
     "@bazel/typescript": "0.37.1",
+    "@types/babel__core": "7.1.3",
```
Comment (alan-agius4) [inline package.json]: Why is this needed? I cannot find any usage of `babel-core`

File: packages/angular_devkit/build_angular/src/browser/index.ts [hunk 3]
```diff
@@ -162,18 +162,61 @@ function getCompilerConfig(wco: WebpackConfigOptions): webpack.Configuration {
...
+  projectSourceRoot?: string;
+  i18n: I18nOptions;
+}> {
   const { config, projectRoot, projectSourceRoot } = await buildBrowserWebpackConfigFromContext(
     options,
     context,
     host,
   );
 
+  // target is verified in the above call
```
[the remark this answers is not in the document]
Comment (clydin) [reply packages/angular_devkit/build_angular/src/browser/index.ts]: Done.

File: packages/angular_devkit/build_angular/src/browser/index.ts (hunk 1 above)
Comment (clydin) [reply packages/angular_devkit/build_angular/src/browser/index.ts]: Done.  Fixed all occurrences.
